lib/Dialect/MSFT/MSFTPasses.cpp

```diff
@@ -146,6 +173,14 @@ void LowerToHWPass::runOnOperation() {
   auto top = getOperation();
   auto *ctxt = &getContext();
 
+  // Traverse MSFT location attributes and export the required Tcl into
+  // templated `sv::VerbatimOp`s with symbolic references to the instance paths.
+  hw::SymbolCache symCache;
+  populateSymbolCache(top, symCache);
+  for (auto hwmod : top.getBody()->getOps<msft::MSFTModuleOp>())
```

Comment:
Rather than relying on the devdb being empty, plz use a pass option containing a list of 'tops'. This could be a separate PR.

Reply:
Got it, will do.

Reply:
@teqdruid I implemented this on the current PR, since I didn't want to land the new `empty` method just to remove it later. From PyCDE, we pass use the list maintained in `self.symbols`, is that what you had in mind?